presto-main/src/main/java/io/prestosql/sql/planner/SymbolsExtractor.java

```diff
@@ -124,7 +125,14 @@ private SymbolsExtractor() {}
     public static Set<QualifiedName> extractNames(Expression expression, Set<NodeRef<Expression>> columnReferences)
     {
         ImmutableSet.Builder<QualifiedName> builder = ImmutableSet.builder();
-        new QualifiedNameBuilderVisitor(columnReferences).process(expression, builder);
+        new QualifiedNameBuilderVisitor(columnReferences, true).process(expression, builder);
+        return builder.build();
+    }
+
+    public static Set<QualifiedName> extractNamesNoSubqueries(Expression expression, Set<NodeRef<Expression>> columnReferences)
+    {
+        ImmutableSet.Builder<QualifiedName> builder = ImmutableSet.builder();
+        new QualifiedNameBuilderVisitor(columnReferences, false).process(expression, builder);
```

Comment:
consider using `io.prestosql.sql.util.AstUtils#preOrder` instead of `DefaultExpressionTraversalVisitor`

Reply:
That's more complicated, as there's no way to stop the recursion with `preOrder`. In any case, I just extended what's already there. That would be a separate cleanup task.